nextmv/configuration.py

```diff
@@ -0,0 +1,186 @@
+"""Configuration for a run."""
+
+import argparse
+import os
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from nextmv.base_model import BaseModel
+
+
+@dataclass
+class Parameter:
+    """
+    Parameter that is used in a `Configuration`. When a parameter is required,
+    it is a good practice to provide a default value for it. This is because
+    the configuration will raise an error if a required parameter is not
+    provided through a command-line argument, an environment variable or a
+    default value.
+    """
+
+    name: str
+    """The name of the parameter."""
+    param_type: type
+    """The type of the parameter."""
+
+    default: Optional[Any] = None
+    """The default value of the parameter. Even though this is optional, it is
+    recommended to provide a default value for all parameters."""
+    description: Optional[str] = None
+    """An optional description of the parameter. This is useful for generating
+    help messages for the configuration."""
+    required: bool = False
+    """Whether the parameter is required. If a parameter is required, it will
+    be an error to not provide a value for it, either trough a command-line
+    argument, an environment variable or a default value."""
+
+
+class Configuration:
```

Comment:
Should this be called configuration? I think we use the term already for the parent of execution class. :thinking: